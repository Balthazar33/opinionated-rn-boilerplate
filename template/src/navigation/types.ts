import {NavigationProp, ParamListBase, RouteProp, StackActionHelpers, StackNavigationState} from '@react-navigation/native';
import {NativeStackNavigationEventMap, NativeStackNavigationOptions} from 'react-native-screens/lib/typescript/native-stack/types';

export enum StackScreens {
  DummyScreen = 'DummyScreen',
  DummyScreenWithScroll = 'DummyScreenWithScroll',
  ApiCallScreen = 'ApiCallScreen',
  PokemonDetailScreen = 'PokemonDetailScreen',
}

export type NativeStackNavigationProp<
  ParamList extends ParamListBase,
  RouteName extends keyof ParamList = string,
  NavigatorID extends string | undefined = undefined
> = NavigationProp<
  ParamList,
  RouteName,
  NavigatorID,
  StackNavigationState<ParamList>,
  NativeStackNavigationOptions,
  NativeStackNavigationEventMap
> &
  StackActionHelpers<ParamList>;

export type NativeStackScreenProps<
  ParamList extends ParamListBase,
  RouteName extends keyof ParamList = string,
  NavigatorID extends string | undefined = undefined
> = {
  navigation: NativeStackNavigationProp<ParamList, RouteName, NavigatorID>;
  route: RouteProp<ParamList, RouteName>;
};

export type RootStackParamList = {
  /**
   * Add screens in the following format:
   * <screen name>: <the param(s) it expects>.
   * Specify 'undefined' if the screen does not expect any params.
   */
  DummyScreen: undefined;
  DummyScreenWithScroll: undefined;
  ApiCallScreen: undefined;
  PokemonDetailScreen: {name: string};
  // add more screens here...
};
