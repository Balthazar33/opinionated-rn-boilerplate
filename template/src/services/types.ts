import {AppDispatch} from '@redux/store.utils';

export interface ApiCallWithErrorHandling {
  apiCall: (params: any) => any;
  params: any;
  dispatch: AppDispatch;
}

export interface ApiCallResponse {
  data: null | any;
  error: {
    message?: string;
  };
}

interface Meta {
  headers?: any;
  status: number;
  statusText: string;
}

export interface CommonResponse {
  data: any;
  meta?: {
    response?: Meta | undefined;
  }; // API call meta data
}

export enum FetchErrorStatus {
  PARSING_ERROR = 'PARSING_ERROR',
}

export enum FetchErrorCodes {
  NOT_FOUND = 404,
  UNAUTHORIZED = 401,
  INCORRECT_INPUT = 400,
  SERVER_ERROR = 500,
}

export interface ApiError {
  data?: any;
  originalStatus: FetchErrorCodes | number;
  status: FetchErrorStatus | string;
}

export interface ServerError extends ApiCallWithErrorHandling {
  error: any;
  errorCode?: ErrorType;
}

export enum StatusCodes {
  ERROR400 = 400,
  ERROR401 = 401,
  ERROR404 = 404,
  ERROR500 = 500,
  ERROR300 = 300,
  SUCCESS200 = 200,
  SUCCESS201 = 201,
}

export type ErrorType =
  | StatusCodes.ERROR400
  | StatusCodes.ERROR401
  | StatusCodes.ERROR404
  | StatusCodes.ERROR500;
