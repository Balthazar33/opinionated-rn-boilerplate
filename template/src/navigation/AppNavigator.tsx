import React from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

import {
  ApiCallScreen,
  DummyScreen,
  DummyScreenWithScroll,
  PokemonDetailScreen,
} from '../screens';
import {RootStackParamList, StackScreens} from './types';

const Stack = createStackNavigator<RootStackParamList>();

function AppNavigator() {
  const commonScreenProps = {
    options: {
      headerShown: false,
    },
  };

  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          {...commonScreenProps}
          name={StackScreens.DummyScreen}
          component={DummyScreen}
        />
        <Stack.Screen
          {...commonScreenProps}
          name={StackScreens.DummyScreenWithScroll}
          component={DummyScreenWithScroll}
        />
        <Stack.Screen
          {...commonScreenProps}
          name={StackScreens.ApiCallScreen}
          component={ApiCallScreen}
        />
        <Stack.Screen
          {...commonScreenProps}
          name={StackScreens.PokemonDetailScreen}
          component={PokemonDetailScreen}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default AppNavigator;
