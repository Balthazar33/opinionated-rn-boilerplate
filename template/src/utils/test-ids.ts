export enum TestIds {
  APP_SAFEAREA = 'AppSafeArea',
  APP_ACTIVITY_INDICATOR = 'ActivityIndicator',
  NO_INTERNET = 'NoInternet',
  TO_API_SCREEN_BTN = 'DummyScreen:ToApiCallScreenBtn',
  DUMMYSCREEN = 'DummyScreen',
  APICALL_SCREEN = 'ApiCallScreen',
}
