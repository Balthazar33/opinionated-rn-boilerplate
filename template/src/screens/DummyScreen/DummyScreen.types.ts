import {RootStackParamList, StackScreens} from '@/navigation/types';
import {StackScreenProps} from '@react-navigation/stack';

export type DummyScreenProps = StackScreenProps<
  RootStackParamList,
  StackScreens.DummyScreen
>;
