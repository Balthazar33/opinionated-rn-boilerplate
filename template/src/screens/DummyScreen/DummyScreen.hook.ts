import {useCallback, useEffect, useMemo, useRef, useState} from 'react';

import {sizer} from '@/utils/metrics';
import {Strings} from '@/utils/strings';
import {commonStyles} from '@/utils/common-styles';
import {FormElementTypes} from '@/components/Form/Form.constants';
import {FormElementProps, FormRef} from '@/components/Form/Form.types';

export const useDummyScreen = () => {
  const formRef = useRef<FormRef>(null);
  const [email, setEmail] = useState('');
  const [username, setUsername] = useState('');
  const [contactNo, setContactNo] = useState('');

  // Simulated error for error boundary
  const [showError, setShowError] = useState(false);
  useEffect(() => {
    return () => {
      setShowError(false);
    };
  }, []);
  const throwError = useCallback(() => {
    setShowError(true);
  }, []);

  const formData = useMemo(() => {
    return {
      [Strings.email]: email,
      [Strings.username]: username,
      [Strings.contact]: contactNo,
    };
  }, [username, contactNo, email]);

  const propsForFormElements = useMemo<
    (FormElementProps | FormElementProps[])[]
  >(() => {
    return [
      {
        maxLength: 30,
        value: username,
        label: Strings.username,
        onChangeText: setUsername,
        fieldName: Strings.username,
        placeholder: '',
        type: FormElementTypes.TEXT_INPUT,
        style: {
          padding: sizer(8),
        },
      },
      [
        {
          maxLength: 10,
          value: contactNo,
          keyboardType: 'numeric',
          label: Strings.contactNo,
          onChangeText: setContactNo,
          fieldName: Strings.contact,
          placeholder: '',
          type: FormElementTypes.TEXT_INPUT,
          style: {
            padding: sizer(8),
          },
          containerStyle: commonStyles.flex1,
        },
        {
          maxLength: 25,
          value: email,
          label: Strings.email,
          onChangeText: setEmail,
          fieldName: Strings.email,
          placeholder: '',
          type: FormElementTypes.TEXT_INPUT,
          style: {
            padding: sizer(8),
          },
          containerStyle: commonStyles.flex1,
        },
      ],
    ];
  }, [username, contactNo, email]);

  return {
    formRef,
    formData,
    showError,
    throwError,
    propsForFormElements,
  };
};
