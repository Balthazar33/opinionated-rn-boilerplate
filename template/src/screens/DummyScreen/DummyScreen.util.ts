import {Strings} from '@/utils/strings';
import {object, string} from 'yup';

export const dummySchema = object({
  [Strings.username]: string()
    .required('Username may not be empty')
    .min(4, 'Username must contain at least 4 characters'),
  [Strings.contact]: string()
    .required('Contact no. may not be empty')
    .min(10, 'Contact no. must be at least 10 characters long'),
  [Strings.email]: string()
    .required('Email may not be empty')
    .email('Please enter a valid email address'),
});
