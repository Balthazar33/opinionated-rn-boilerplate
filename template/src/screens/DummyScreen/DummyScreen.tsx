import React, {useCallback} from 'react';
import {StyleSheet, View} from 'react-native';

import {TouchableRipple} from 'react-native-paper';

import {sizer} from '@utils/metrics';
import {Strings} from '@utils/strings';
import {TestIds} from '@utils/test-ids';
import {SCREEN_PADDING} from '@utils/constants';
import {StackScreens} from '@/navigation/types';
import {DummyScreenProps} from './DummyScreen.types';
import {ScrollableScreen} from '@/containers/BaseScreen';
import {TextBold16, TextRegular14} from '@components/Typography';
import {Form} from '@/components/Form/Form';
import {useDummyScreen} from './DummyScreen.hook';
import {dummySchema} from './DummyScreen.util';
import {Colors} from '@/utils/colors';

export const DummyScreen = ({navigation}: DummyScreenProps) => {
  const {formData, propsForFormElements, formRef, showError, throwError} =
    useDummyScreen();

  const handleBtnPress = useCallback(() => {
    navigation.navigate(StackScreens.ApiCallScreen);
  }, [navigation]);

  const handleValidatePress = useCallback(() => {
    const isValid = formRef?.current?.validate?.();
    console.warn(`The form is ${isValid ? 'valid' : 'invalid'}`);
  }, [formRef]);

  if (showError) {
    throw new Error('This is not a drill!');
  }
  return (
    <ScrollableScreen testID={TestIds.DUMMYSCREEN} style={style.screenStyle}>
      <TouchableRipple
        rippleColor={Colors.GREY}
        onPress={handleBtnPress}
        style={style.buttonStyle}
        testID={TestIds.TO_API_SCREEN_BTN}>
        <TextRegular14 color={Colors.WHITE}>
          {Strings.toApiCallScreen}
        </TextRegular14>
      </TouchableRipple>
      <TouchableRipple
        rippleColor={Colors.GREY}
        onPress={throwError}
        style={style.buttonStyle}>
        <TextRegular14 color={Colors.WHITE}>{Strings.throwError}</TextRegular14>
      </TouchableRipple>
      {/* ---------------Form example--------------- */}
      <View style={style.formContainer}>
        <TextBold16>Form</TextBold16>
        <Form
          ref={formRef}
          formData={formData}
          schema={dummySchema}
          formElementProps={propsForFormElements}
        />
        <TouchableRipple
          rippleColor={Colors.GREY}
          onPress={handleValidatePress}
          style={style.buttonStyle}>
          <TextRegular14 style={style.textCenter} color={Colors.WHITE}>
            Validate form
          </TextRegular14>
        </TouchableRipple>
      </View>
      {/* ---------------Form example--------------- */}
    </ScrollableScreen>
  );
};

const style = StyleSheet.create({
  screenStyle: {
    gap: sizer(16),
    padding: SCREEN_PADDING,
  },
  formContainer: {
    gap: 16,
    marginTop: 16,
  },
  textCenter: {
    textAlign: 'center',
  },
  buttonStyle: {
    padding: sizer(16),
    backgroundColor: Colors.PRIMARY_BLUE,
  },
});
