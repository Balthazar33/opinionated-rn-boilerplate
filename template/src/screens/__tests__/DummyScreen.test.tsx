import React from 'react';

import renderer from 'react-test-renderer';
import {it, describe, expect} from '@jest/globals';
import '@testing-library/react-native/extend-expect';
import {fireEvent, render} from '@testing-library/react-native';

import {DummyScreen} from '..';
import {TestIds} from '../../utils/test-ids';
import {DummyScreenProps} from '../DummyScreen/DummyScreen.types';
import {StackScreens} from '../../navigation/types';

const navigation: Partial<DummyScreenProps['navigation']> = {
  navigate: jest.fn,
};
const route: Partial<DummyScreenProps['route']> = {
  params: undefined,
};

describe('DummyScreen', () => {
  it('renders correctly', () => {
    const elementTree = renderer
      .create(
        <DummyScreen
          navigation={navigation as DummyScreenProps['navigation']}
          route={route as DummyScreenProps['route']}
        />,
      )
      .toJSON();
    expect(elementTree).toMatchSnapshot();
  });

  it('should navigate to API Call screen on pressing "To Api Call Screen" button', () => {
    const navigateSpy = jest.spyOn(navigation, 'navigate');
    const {getByTestId} = render(
      <DummyScreen
        navigation={navigation as DummyScreenProps['navigation']}
        route={route as DummyScreenProps['route']}
      />,
    );
    const toApiCallScreenBtn = getByTestId(TestIds.TO_API_SCREEN_BTN);
    fireEvent(toApiCallScreenBtn, 'press');
    expect(navigateSpy).toHaveBeenCalledWith(StackScreens.ApiCallScreen);
  });
});
