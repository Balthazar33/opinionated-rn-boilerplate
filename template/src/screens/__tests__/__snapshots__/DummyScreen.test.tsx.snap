// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyScreen renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  keyboardShouldPersistTaps="always"
>
  <View>
    <RCTSafeAreaView
      style={
        {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      testID="AppSafeArea"
    >
      <View
        style={
          [
            {
              "flex": 1,
            },
            {
              "gap": 32,
              "padding": 32,
            },
          ]
        }
        testID="Screens:DummyScreen"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              false,
              {
                "backgroundColor": "#0000ff",
                "padding": 32,
              },
            ]
          }
          testID="DummyScreen:ToApiCallScreenBtn"
        >
          <Text
            color="#ffffff"
            style={
              {
                "color": "#ffffff",
                "fontFamily": "roboto",
                "fontSize": 28,
              }
            }
          >
            To Api Call Screen
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              false,
              {
                "backgroundColor": "#0000ff",
                "padding": 32,
              },
            ]
          }
        >
          <Text
            color="#ffffff"
            style={
              {
                "color": "#ffffff",
                "fontFamily": "roboto",
                "fontSize": 28,
              }
            }
          >
            Throw Error
          </Text>
        </View>
        <View
          style={
            {
              "gap": 16,
              "marginTop": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontFamily": "roboto",
                "fontSize": 32,
              }
            }
          >
            Form
          </Text>
          <View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(231, 224, 236, 1)",
                    "borderTopLeftRadius": 4,
                    "borderTopRightRadius": 4,
                  },
                  {
                    "backgroundColor": "#8a8a8a30",
                    "fontFamily": "roboto",
                    "padding": 16,
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "rgba(73, 69, 79, 1)",
                    "bottom": 0,
                    "height": 1,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "transform": [
                      {
                        "scaleY": 0.5,
                      },
                    ],
                    "zIndex": 1,
                  }
                }
                testID="text-input-underline"
              />
              <View
                onLayout={[Function]}
                style={
                  [
                    {
                      "flexGrow": 1,
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    {
                      "minHeight": 56,
                    },
                  ]
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    [
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      {
                        "overflow": "hidden",
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                        ],
                        "width": 750,
                        "zIndex": 3,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "width": 73,
                        }
                      }
                    >
                      <Text
                        collapsable={false}
                        maxFontSizeMultiplier={1.5}
                        numberOfLines={1}
                        onLayout={[Function]}
                        onTextLayout={[Function]}
                        style={
                          {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontSize": 28,
                            "fontWeight": undefined,
                            "left": 0,
                            "letterSpacing": 0.15,
                            "lineHeight": undefined,
                            "maxWidth": 73,
                            "opacity": 0,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                            "position": "absolute",
                            "textAlign": "left",
                            "top": 30,
                            "transform": [
                              {
                                "translateX": 0,
                              },
                              {
                                "translateY": 0,
                              },
                              {
                                "scale": 1,
                              },
                            ],
                            "writingDirection": "ltr",
                          }
                        }
                        testID="text-input-flat-label-active"
                      >
                        Username
                      </Text>
                      <Text
                        collapsable={false}
                        maxFontSizeMultiplier={1.5}
                        numberOfLines={1}
                        style={
                          {
                            "color": "rgba(73, 69, 79, 1)",
                            "fontFamily": "System",
                            "fontSize": 28,
                            "fontWeight": undefined,
                            "left": 0,
                            "letterSpacing": 0.15,
                            "lineHeight": undefined,
                            "maxWidth": 73,
                            "opacity": 0,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                            "position": "absolute",
                            "textAlign": "left",
                            "top": 30,
                            "transform": [
                              {
                                "translateX": 0,
                              },
                              {
                                "translateY": 0,
                              },
                              {
                                "scale": 1,
                              },
                            ],
                            "writingDirection": "ltr",
                          }
                        }
                        testID="text-input-flat-label-inactive"
                      >
                        Username
                      </Text>
                    </View>
                  </View>
                </View>
                <TextInput
                  cursorColor="#000000"
                  editable={true}
                  maxFontSizeMultiplier={1.5}
                  maxLength={30}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  outlineColor="#8a8a8a"
                  placeholder=" "
                  placeholderTextColor="#8a8a8a"
                  selectionColor="#000000"
                  style={
                    [
                      {
                        "flexGrow": 1,
                        "margin": 0,
                      },
                      {},
                      {
                        "paddingBottom": 4,
                        "paddingTop": 24,
                      },
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 28,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "minWidth": 65,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-flat"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "gap": 8,
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "backgroundColor": "rgba(231, 224, 236, 1)",
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    {
                      "backgroundColor": "#8a8a8a30",
                      "fontFamily": "roboto",
                      "padding": 16,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "rgba(73, 69, 79, 1)",
                      "bottom": 0,
                      "height": 1,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": [
                        {
                          "scaleY": 0.5,
                        },
                      ],
                      "zIndex": 1,
                    }
                  }
                  testID="text-input-underline"
                />
                <View
                  onLayout={[Function]}
                  style={
                    [
                      {
                        "flexGrow": 1,
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "minHeight": 56,
                      },
                    ]
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "overflow": "hidden",
                        },
                      ]
                    }
                  >
                    <View
                      collapsable={false}
                      pointerEvents="none"
                      style={
                        {
                          "bottom": 0,
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                          "width": 750,
                          "zIndex": 3,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "width": 73,
                          }
                        }
                      >
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          onLayout={[Function]}
                          onTextLayout={[Function]}
                          style={
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 28,
                              "fontWeight": undefined,
                              "left": 0,
                              "letterSpacing": 0.15,
                              "lineHeight": undefined,
                              "maxWidth": 73,
                              "opacity": 0,
                              "paddingLeft": 16,
                              "paddingRight": 16,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 30,
                              "transform": [
                                {
                                  "translateX": 0,
                                },
                                {
                                  "translateY": 0,
                                },
                                {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                          testID="text-input-flat-label-active"
                        >
                          Contact No.
                        </Text>
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          style={
                            {
                              "color": "rgba(73, 69, 79, 1)",
                              "fontFamily": "System",
                              "fontSize": 28,
                              "fontWeight": undefined,
                              "left": 0,
                              "letterSpacing": 0.15,
                              "lineHeight": undefined,
                              "maxWidth": 73,
                              "opacity": 0,
                              "paddingLeft": 16,
                              "paddingRight": 16,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 30,
                              "transform": [
                                {
                                  "translateX": 0,
                                },
                                {
                                  "translateY": 0,
                                },
                                {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                          testID="text-input-flat-label-inactive"
                        >
                          Contact No.
                        </Text>
                      </View>
                    </View>
                  </View>
                  <TextInput
                    cursorColor="#000000"
                    editable={true}
                    keyboardType="numeric"
                    maxFontSizeMultiplier={1.5}
                    maxLength={10}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    outlineColor="#8a8a8a"
                    placeholder=" "
                    placeholderTextColor="#8a8a8a"
                    selectionColor="#000000"
                    style={
                      [
                        {
                          "flexGrow": 1,
                          "margin": 0,
                        },
                        {},
                        {
                          "paddingBottom": 4,
                          "paddingTop": 24,
                        },
                        {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 28,
                          "fontWeight": undefined,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "minWidth": 65,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        [
                          {},
                        ],
                        undefined,
                      ]
                    }
                    testID="text-input-flat"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "backgroundColor": "rgba(231, 224, 236, 1)",
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                    },
                    {
                      "backgroundColor": "#8a8a8a30",
                      "fontFamily": "roboto",
                      "padding": 16,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "rgba(73, 69, 79, 1)",
                      "bottom": 0,
                      "height": 1,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": [
                        {
                          "scaleY": 0.5,
                        },
                      ],
                      "zIndex": 1,
                    }
                  }
                  testID="text-input-underline"
                />
                <View
                  onLayout={[Function]}
                  style={
                    [
                      {
                        "flexGrow": 1,
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "minHeight": 56,
                      },
                    ]
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "overflow": "hidden",
                        },
                      ]
                    }
                  >
                    <View
                      collapsable={false}
                      pointerEvents="none"
                      style={
                        {
                          "bottom": 0,
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "transform": [
                            {
                              "translateX": 0,
                            },
                          ],
                          "width": 750,
                          "zIndex": 3,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "width": 73,
                          }
                        }
                      >
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          onLayout={[Function]}
                          onTextLayout={[Function]}
                          style={
                            {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 28,
                              "fontWeight": undefined,
                              "left": 0,
                              "letterSpacing": 0.15,
                              "lineHeight": undefined,
                              "maxWidth": 73,
                              "opacity": 0,
                              "paddingLeft": 16,
                              "paddingRight": 16,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 30,
                              "transform": [
                                {
                                  "translateX": 0,
                                },
                                {
                                  "translateY": 0,
                                },
                                {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                          testID="text-input-flat-label-active"
                        >
                          Email
                        </Text>
                        <Text
                          collapsable={false}
                          maxFontSizeMultiplier={1.5}
                          numberOfLines={1}
                          style={
                            {
                              "color": "rgba(73, 69, 79, 1)",
                              "fontFamily": "System",
                              "fontSize": 28,
                              "fontWeight": undefined,
                              "left": 0,
                              "letterSpacing": 0.15,
                              "lineHeight": undefined,
                              "maxWidth": 73,
                              "opacity": 0,
                              "paddingLeft": 16,
                              "paddingRight": 16,
                              "position": "absolute",
                              "textAlign": "left",
                              "top": 30,
                              "transform": [
                                {
                                  "translateX": 0,
                                },
                                {
                                  "translateY": 0,
                                },
                                {
                                  "scale": 1,
                                },
                              ],
                              "writingDirection": "ltr",
                            }
                          }
                          testID="text-input-flat-label-inactive"
                        >
                          Email
                        </Text>
                      </View>
                    </View>
                  </View>
                  <TextInput
                    cursorColor="#000000"
                    editable={true}
                    maxFontSizeMultiplier={1.5}
                    maxLength={25}
                    multiline={false}
                    numberOfLines={1}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    outlineColor="#8a8a8a"
                    placeholder=" "
                    placeholderTextColor="#8a8a8a"
                    selectionColor="#000000"
                    style={
                      [
                        {
                          "flexGrow": 1,
                          "margin": 0,
                        },
                        {},
                        {
                          "paddingBottom": 4,
                          "paddingTop": 24,
                        },
                        {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 28,
                          "fontWeight": undefined,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "minWidth": 65,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        [
                          {},
                        ],
                        undefined,
                      ]
                    }
                    testID="text-input-flat"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                {
                  "backgroundColor": "#0000ff",
                  "padding": 32,
                },
              ]
            }
          >
            <Text
              color="#ffffff"
              style={
                [
                  {
                    "color": "#ffffff",
                    "fontFamily": "roboto",
                    "fontSize": 28,
                  },
                  {
                    "textAlign": "center",
                  },
                ]
              }
            >
              Validate form
            </Text>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;
